doc/getting-started.rst

```diff
@@ -0,0 +1,565 @@
+name that names the software to build the part and optionally provides
+parameters to control how the part is built.
+
+Installing software
+===================
+
+In this tutorial, we're going to install a simple database server.
+The details of the server aren't important.  It just provides a useful
+example that illustrates a number of ways that Buildout can make
+things easier.
+
+We'll start by adding a part to install the server software.  We'll
+update our Buildout configuration to add a ``zeo`` part::
+
+  [buildout]
+  parts = zeo
+
+  [zeo]
+  recipe = zc.recipe.egg
+  eggs = ZEO
+
+We added the part name, ``zeo`` to the ``parts`` option in the
+``buildout`` section.  We also added a ``zeo`` section with two
+options:
+
+recipe
+  The standard ``recipe`` option names the software component that
+  will implement the part.  The value is a Python distribution
+  requirement, as would be used with ``pip``.  In this case, we've
+  specified `zc.recipe.egg
+  <https://pypi.python.org/pypi/zc.recipe.egg>`_ which is the name of
+  a Python project that provides a number of recipe implementations.
+
+eggs
+  A list of distribution requirements, one per
+  line. [#requirements-one-per-line]_ (The name of this option is
+  unfortunate, because the values are requirements, not egg names.)
+  Listed requirements are installed, along with their dependencies. In
+  addition, any scripts provided by the listed requirements (but not
+  their dependencies) are installed in the ``bin`` directory.
+
+If we run this [#gcc]_::
+
+  buildout
+
+Then a number of things will happen:
+
+- ``zc.recipe.egg`` will be downloaded and installed in your ``eggs``
+  directory.
+
+- ``ZEO`` and its dependencies will be downloaded and installed. (ZEO
+  is a small Python database server.)
+
+- A number of scripts will be installed in the ``bin`` directory.  One
+  in particular, ``runzeo`` is used to run a ZEO server.
+
+Generating configuration and custom scripts
+===========================================
+
+The ``runzeo`` program doesn't daemonize itself. Rather, it's meant to
+be used with a dedicated daemonizer like `zdaemon
+<https://pypi.python.org/pypi/zdaemon>`_ or `supervisord
+<http://supervisord.org/>`_.  We'll use a `recipe to set up zdaemon
+<https://pypi.python.org/pypi/zc.zdaemonrecipe>`_.  Our Buildout
+configuration becomes::
+
+  [buildout]
+  parts = zeo server
+
+  [zeo]
+  recipe = zc.recipe.egg
+  eggs = ZEO
+
+  [server]
+  recipe = zc.zdaemonrecipe
+  program =
+    ${buildout:bin-directory}/runzeo
+      -f ${buildout:directory}/data.fs
+      -a 127.0.0.1:8200
+
+Here we've added a new ``server`` part that uses ``zc.zdaemonrecipe``.
+We used a ``program`` option to define what program should be run.
+There are a couple of interesting things to note about this option:
+
+- We used :doc:`variable substitutions
+  <topics/variables-extending-and-substitutions>`:
+
+  ``${buildout:directory}``
```

Comment:
No. :)  

I think in a document like this, I'd rather keep things simple and goal oriented.  We can't explain everything at once. I haven't explained what the significance of ``buildout::`` is so there's no reason for people to wonder why these aren't defined there. I provide a link to a topic where they can get the details.

These aren't internal variables. They're part of Buildout's API and will be documented in the reference.  The variables-extending-and-substitutions will mention that variables have defaults (among many other related things).

I added a sentence referring them to the reference to find out what settings are available.